File: frontend/src/api/mlflow.ts
import { fetchWithAuth } from "./client";

export type ModelRegistry = {
  id: number;
  version: string;
  algorithm: string;
  accuracy: number;
  f1_score: number;
  status: string;
  created_at: string;
};

export type ModelRegistryListResponse = {
  items: ModelRegistry[];
};

export async function getModels(): Promise<ModelRegistryListResponse> {
  const res = await fetchWithAuth(`/models`);
  if (!res.ok) {
    const error = await res.json();
    throw new Error(error.detail || "Failed to fetch models");
  }
  return res.json();
}

export async function promoteModel(modelId: number): Promise<ModelRegistry> {
  const res = await fetchWithAuth(`/models/${modelId}/promote`, {
    method: "POST"
  });
  if (!res.ok) {
    const error = await res.json();
    throw new Error(error.detail || "Failed to promote model");
  }
  return res.json();
}

File: backend/app/schemas/mlflow.py
from pydantic import BaseModel
from typing import List
from datetime import datetime

class ModelRegistryResponse(BaseModel):
    id: int
    version: str
    algorithm: str
    accuracy: float
    f1_score: float
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class ModelRegistryListResponse(BaseModel):
    items: List[ModelRegistryResponse]
